pollux/core/src/main/scala/org/hyperledger/identus/pollux/core/service/uriResolvers/DidUrlResolver.scala

```diff
@@ -0,0 +1,132 @@
+package org.hyperledger.identus.pollux.core.service.uriResolvers
+
+import io.lemonlabs.uri.{Url, UrlPath}
+import org.hyperledger.identus.pollux.vc.jwt
+import org.hyperledger.identus.pollux.vc.jwt.*
+import org.hyperledger.identus.shared.crypto.Sha256Hash
+import org.hyperledger.identus.shared.http.{GenericUriResolverError, UriResolver}
+import org.hyperledger.identus.shared.models.PrismEnvelopeData
+import org.hyperledger.identus.shared.models.StatusCode
+import zio.*
+import zio.json.*
+
+class DidUrlResolver(httpUrlResolver: HttpUrlResolver, didResolver: DidResolver) extends UriResolver {
+  import DidUrlResolver.*
+
+  def resolve(uri: String): IO[GenericUriResolverError, String] = {
```

Comment:
I would also try to refactor the long function and split it into the stages/phases:
1. parse Uri => get DID
2. resolve DID
3. fetch Services
4. compose the resource Url
5. get the resource by Url
6. check resource integrity (if needed)